docs/proposals/Ashish_Mahajan_Versioning_Proposal.md

```diff
@@ -0,0 +1,241 @@
+
+1. **Hidden Versions**: Users can't tell which version they're viewing or that other versions exist, so they might miss updated recommendations.
+
+2. **Hard to Navigate**: Switching between versions requires typing different URLs or navigating through GitHub, making for a poor user experience.
+
+3. **Content Evolution Opacity**: Users can't easily see how DEI recommendations have changed over time.
+
+
+### Solution Architecture
+
+My implementation follows the industry-standard branch-based versioning model used by scikit-learn, Matplotlib, NetworkX, and other NumFOCUS projects:
+
+1. **Branch-Version Mapping**:
+   - main branch (v2.0) → deployed to / (root path)
+   - master branch (v1.0) → deployed to /v1.0/
+   - future version branches → deployed to /v{X.Y}/
+
+2. **URL Structure**:
+   - Latest version: discover-cookbook.numfocus.org/
+   - Previous version: discover-cookbook.numfocus.org/v1.0/
+   - Version-specific pages: discover-cookbook.numfocus.org/v1.0/04_venue_selection/
+
+3. **Version UI Components**:
+   - Version dropdown menu in the left sidebar.
+   - Clear current version indicator
+   - Outdated version notification for non-latest versions(via banner at top)
+   - Context-preserving navigation between versions
+
+### User Benefits
+
+This implementation delivers significant user experience improvements:
+
+1. **Clear Version Context**: Users immediately know which version they're viewing and what other versions exist.
+
+2. **Seamless Navigation**: One-click switching between versions preserves topic context when possible.
+
+3. **Historical Transparency**: Easy access to how DEI recommendations have evolved over time.
+
+
+### User Scenarios
+
+**Scenario 1: Conference Organizer - Version Comparison**
+Sunita is organizing a conference and needs to compare current and previous recommendations:
+
+- She visits discover-cookbook.numfocus.org and sees she's viewing v2.0
+- She notices the version selector in the navigation
+- She clicks to view v1.0 while viewing the "venue selection" section
+- The system navigates her to the equivalent section in v1.0
+- She can easily toggle between versions to compare approaches
+<br><br>
+![version-selector](togglebutton1.png)
+<br><br>
+
+**Scenario 2: First-time Visitor - Version Awareness**
+Karan discovers the cookbook through social media:
+
+- He lands on discover-cookbook.numfocus.org (latest version)
+- The version indicator shows he's viewing the current version (v2.0)
+- He's aware previous versions exist through the version selector
+- If he follows an old link to v1.0, he'll see a banner indicating newer content exists
+
+![Banner](banner.png)
+
+## Implementation
+
+My implementation follows a systematic approach aligned with industry standards for documentation versioning:
+
+### 1. Version Data Architecture
+
+I will implement a centralized version registry that defines all versions and their relationships. This architecture will:
```

Comment:
no this would be too complex to manage. We really want to just have tags for the "editions" like a version of a software library.